### ISSUE TITLE
Simple linear and non-linear dependencies

### ISSUE THREAD

**Issue by ccrnn** [dynamicslab/pysindy#253]
I was thinking that pysindy could also be used to identify and remove columns which have simple linear or non-linear relationships (or close enough) from a larger dataset, in order to reduce the dataset to try to get a simpler dynamic system.

It seems like there will be better ways of doing that, but it makes me wonder if there's something that can be improved which would make these simple relationships easier for SINDy to figure out, which might improve model fitting in general too?

I also noticed that the first term is sometimes kind of confusing, here c' looks like multiplication by 2 when it should be a^2, though the model is fitting correctly:
(a)' = 1.000 1
(b)' = 2.000 1 + -2.000 a + 1.000 b
(c)' = 2.000 a

```
from diffeqpy import de
import numpy as np
import pysindy as ps
import matplotlib.pyplot as plt

a2 = np.array([[1,2,3,4,5,6,7,8],
                [2,4,6,8,10,12,14,17],
                [1,4,9,16,25,36,49,64]])

t = np.array([0,1,2,3,4,5,6,7])
t = np.transpose(t)
ut = np.transpose(a2)

feature_names = ['a', 'b', 'c']
opt = ps.STLSQ(threshold=0.1)
fl = ps.PolynomialLibrary(degree=2)
model = ps.SINDy(feature_names=feature_names,optimizer=opt, feature_library=fl)
f = model.fit(ut, t)
model.print()
y = model.predict(ut)
```
Here b' seems overcomplicated (and becomes much more so if I add extra small changes, but noise can also be dealt with another way).

If I change the series in c to start at 0, the model again becomes unnecessarily complicated:
(a)' = 1.000 1
(b)' = 2.000 1 + -2.000 a + 1.000 b
(c)' = -1.000 1 + -1.000 c + 0.132 a^2 + 0.168 a b + 0.234 a c + 0.133 b^2 + -0.117 b c

Maybe SINDy is set up in a way that gives weight to models which use more of the other variables in the model? Which for many dynamical systems will be helpful / the right thing to do. But it depends on being sure you have the simplest explanatory variables in your data to begin with, and it seems like there may be times SINDy will come out with an overcomplicated model if variables do have such simple relationships...

**Comment by ccrnn** [comment]
I understood how to do this a different way... So no worries, and thanks.

**Comment by ccrnn** [comment]
Actually, I'm not sure I should have closed this - even though I understood a different way of doing what I had intended, it might still be nice if there was an option to make the output more or less heavily dependent on single other variables or a mixture of variables.

Sometimes I also get output with extremely high values for the first figure [1], have I missed an option where I can control this?
Thanks again for the help!

**Comment by Jacob-Stevens-Haas** [comment]
For time-shift variables, you might look at methods like pykoopman or hankel.  I'm not sure though, it's not my area of expertise.

Idk what you mean by "figure [1]"... Did you mean the constant coefficient in the ODEs?

**Comment by ccrnn** [comment]
Thanks, i'll check them out.

Yeah, it's the constant coefficient. Am I always supposed to normalise data or is there an option somewhere that controls for this?

**Comment by Jacob-Stevens-Haas** [comment]
No, if there's a constant in the true ODE, that term should recover it.  If you don't want it, I believe the feature libraries have a `include_bias` constructor argument that can be set to `False` in order to remove the constant term.

**Comment by ccrnn** [comment]
Thanks for the help! I got this to work with normalisation and include_bias = false.
